board_types.txt

```diff
@@ -0,0 +1,33 @@
+# this file lists all board types for boards that use the bootloader
+# protocol implemented in this bootloader. This file should be updated
+# when a new board type is added
+
+TARGET_HW_PX4_FMU_V1                    5
+TARGET_HW_PX4_FMU_V2                    9
+TARGET_HW_PX4_FMU_V3                    9 # same as FMU_V2
+TARGET_HW_PX4_FMU_V4                   11
+TARGET_HW_PX4_FMU_V4_PRO               13
+TARGET_HW_PX4_FMU_V5                   50
+TARGET_HW_MINDPX_V2                    88
+TARGET_HW_PX4_FLOW_V1                   6
+TARGET_HW_PX4_DISCOVERY_V1             99
+TARGET_HW_PX4_PIO_V1                   10
+TARGET_HW_PX4_PIO_V2                   10 # same as PIO_V1
+TARGET_HW_PX4_PIO_V3                   13
+TARGET_HW_PX4_AEROCORE_V1              98
+TARGET_HW_TAP_V1                       64
+TARGET_HW_CRAZYFLIE                    12
+TARGET_HW_OMNIBUSF4SD                  42
+TARGET_HW_AUAV_X2V1                    33
+TARGET_HW_AEROFC_V1                    65
+TARGET_HW_CUBE_F4                       9
+TARGET_HW_CUBE_F7                      50
+
+
+# values below this line are implemented in the ArduPilot bootloader
```

Comment:
@tridge 

So that we can keep these in sync better. Can we point to the repo where the AP boot loaded in located in this text file?  Also I have a set to code points that need to be added as well. I would also asks that we keep the TARGET_HW prefix for inter op.